spring-cloud-etcd-discovery/src/main/java/org/springframework/cloud/etcd/discovery/EtcdDiscoveryClientConfiguration.java

```diff
@@ -36,17 +41,20 @@
 	private EtcdClient client;
 
 	@Bean
-	public EtcdLifecycle etcdLifecycle() {
-		return new EtcdLifecycle(client, etcdDiscoveryProperties());
+	@ConditionalOnMissingBean
+	public TtlScheduler ttlScheduler(EtcdDiscoveryProperties properties) {
```

Comment:
Can you explain why this is needed? Etcd has a ttl built in.

Reply:
I thought we would want to mark the service's availability via a heartbeat. What's the alternative?

Reply:
You're right, How about renaming it using `Heartbeat`.

Reply:
Addressed in 601d7b8.